.github/workflows/pipeline.yml

```diff
@@ -0,0 +1,183 @@
+name: Pipeline [test -> build -> deploy]
+
+on:
+  push:
+    branches:
+      - '**'
+  workflow_dispatch:
+    inputs:
+      additional_docker_tag:
+        description: Additional docker tag that can be used to specify stable or testing tags
+        required: false
+        default: ''
+        type: string
+      push:
+        description: Push docker image to registry flag
+        required: true
+        default: false
+        type: boolean
+      node_version_file:
+        description: "Passed to setup-node action to specify where to source the version of node from"
+        required: false
+        type: string
+        default: ".nvmrc"
+
+permissions:
+  contents: read
+  packages: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+# This will cancel all running build/test/release pipelines that are not on the main branch
+# If this pipeline is on the main branch, it will wait until existing runs complete
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  # main node build workflow
+  node_build:
+    name: node build
+    uses: ministryofjustice/hmpps-github-actions/.github/workflows/node_build.yml@v2 # WORKFLOW_VERSION
+    secrets: inherit
+  # generic node unit tests - feel free to override with local tests if required
+  node_unit_tests:
+    name: node unit tests
+    uses: ministryofjustice/hmpps-github-actions/.github/workflows/node_unit_tests.yml@v2 # WORKFLOW_VERSION
+    needs: [node_build]
+    secrets: inherit
+  # generic node integration tests using wiremock - feel free to override with local tests if required
+  node_integration_tests:
+    runs-on: ubuntu-latest
+    needs: [node_build]
+    services:
+      redis:
+        image: redis:7.0
```

Comment:
if redis can be used as a service in external workflows, why can't minio?

Reply:
I tried but it's another limitation of GH Actions where you cannot override the command as we need in the case of MinIO.

Reply:
it’s pretty annoying that GHA doesn’t let you change the container command or even pass arguments :/
`options` is actually for the `docker create …` command!

Reply:
Yep.